packages/ui/react-ui-stack/src/components/StackContext.tsx

```diff
@@ -13,9 +13,9 @@ export type StackItemSize = DndSize;
 
 export type StackItemData = { id: string; type: 'column' | 'card' };
 
-export type StackItemRearrangeHandler = (
-  source: StackItemData,
-  target: StackItemData,
+export type StackItemRearrangeHandler<D extends { id: string } = StackItemData> = (
+  source: D,
+  target: D,
```

Comment:
```suggestion
export type StackItemRearrangeHandler<Data extends { id: string } = StackItemData> = (
  source: Data,
  target: Data,
```

imo no reason type generics need to be a single letter if it helps explain what the generic is